flink-runtime/src/test/java/org/apache/flink/runtime/clusterframework/BootstrapToolsTest.java

```diff
@@ -360,4 +368,32 @@ public void testConcurrentActorSystemCreation() throws Exception {
 			ExecutorUtils.gracefulShutdown(10000L, TimeUnit.MILLISECONDS, executorService);
 		}
 	}
+
+	/**
+	 * Tests that the {@link ActorSystem} fails with an expressive exception if it cannot be
+	 * instantiated due to an occupied port.
+	 */
+	@Test
+	public void testActorSystemInstantiationFailureWhenPortOccupied() throws Exception {
+		ServerSocket portOccupier;
+		final int port;
+
+		try {
+			port = NetUtils.getAvailablePort();
+			portOccupier = new ServerSocket(port, 10, InetAddress.getByName("0.0.0.0"));
+		}
+		catch (Throwable t) {
+			// could not find free port, or open a connection there
```

Comment:
It's acceptable to silently fail if we cannot open a socket?

Reply:
I think you're right that this not necessary by not preselecting the `port`. Instead we should let the `ServerSocket` pick a port via `ServerSocket(0,...)`. Will change it accordingly.